### PR TITLE
Extract common methods for parser

### LINKED ISSUE #139: Duplicate methods used in multiple parser classes

### PR DESCRIPTION
**Pull request by OVReader**
Fix #139

### CODE REVIEW

File: src/main/java/seedu/duke/parsermanager/Parser.java [hunk 1]
```diff
@@ -1,24 +1,81 @@
...
+
+    protected ArrayList<String> extractCommandDetails(String rawCommandDetail, String[] flags,
+                                                    int[] flagIndexPositions) {
+        ArrayList<String> extractedCommandDetails = new ArrayList<>();
+        for (int i = 0; i < flags.length; i++) {
+            String extractedDetail;
+            if (i == flags.length - UNIT_VALUE) {
+                /* The extracted detail for the last flag starts from the char after the flag, to the end of
+                   rawCommandDetails */
+                extractedDetail = extractDetail(rawCommandDetail, flagIndexPositions[i] + flags[i].length());
```
Comment (wilsonngja) [inline src/main/java/seedu/duke/parsermanager/Parser.java:45]: Perhaps can try to use a different name. The method name and the variable name seems similar and can be confusing.

File: src/main/java/seedu/duke/parsermanager/Parser.java [hunk 2]
```diff
@@ -1,24 +1,81 @@
...
+        ArrayList<Integer> integerDetails = new ArrayList<>();
+        for (String detail : processedCommandDetails) {
+            int integer;
+            try {
+                integer = Integer.parseInt(detail);
+            } catch (NumberFormatException e) {
+                throw new NotIntegerException(EXCEPTION);
+            }
+            // Convert to 0-index
+            integerDetails.add(integer - UNIT_VALUE);
```
Comment (wilsonngja) [inline src/main/java/seedu/duke/parsermanager/Parser.java:77]: Could try to assign a variable to it to make it even more understand.

**Review by wilsonngja — COMMENTED** [review]
Good job in removing repetition. 👍🏼 
There are small coding standard ambiguity but overall LGTM.

**Review by FeliciaBeatrice — APPROVED** [review]
LGTM

**Pull request merged**